pkg/instancegroups/instancegroups.go

```diff
@@ -137,6 +142,13 @@ func (r *RollingUpdateInstanceGroup) RollingUpdate(rollingUpdateData *RollingUpd
 		}
 	}
 
+	if !rollingUpdateData.CloudOnly {
+		err = r.taintAllNeedUpdate(update, rollingUpdateData)
```

Comment:
Tip (that I am a recent convert to): Using `if err := foo(); err != nil { return err }` avoids problems with variable shadowing (though it really confuses the go static analysis tools, and it only works when it's a single error retval!)

Reply:
Thanks, I'll try it out.